request/graphql/schema/collection.go

```diff
@@ -23,8 +23,19 @@ import (
 	"github.com/graphql-go/graphql/language/source"
 )
 
+const (
+	indexDirectiveLabel          = "index"
```

Comment:
todo: I cannot spot where the index directive is declared, and made available to the GQL schema - are you sure this works via a gql client?

todo: Related to the above, I think it is well worth adding schema tests to make sure this works, and continues to do so.

todo: I think this is the wrong place to declare these consts, I would move them to `request/graphql/schema/types.types.go`, alongside the explain directives (a good example of how to declare and make the directive available to the GQL type system).